jetty-deploy/src/test/java/org/eclipse/jetty/deploy/DeploymentTempDirTest.java

```diff
@@ -0,0 +1,231 @@
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.deploy.providers.WebAppProvider;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.ContextHandlerCollection;
+import org.eclipse.jetty.server.handler.DefaultHandler;
+import org.eclipse.jetty.server.handler.HandlerCollection;
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.Scanner;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DeploymentTempDirTest
+{
+    private static final Logger LOG = Log.getLogger(DeploymentTempDirTest.class);
+
+    private final WebAppProvider webAppProvider = new WebAppProvider();
+    private final ContextHandlerCollection contexts = new ContextHandlerCollection();
+    private final Path testDir = MavenTestingUtils.getTargetTestingPath(DeploymentTempDirTest.class.getSimpleName());
+    private final Path tmpDir = testDir.resolve("tmpDir");
+    private final Path webapps = testDir.resolve("webapps");
+    private final Server server = new Server();
+
+    @BeforeEach
+    public void setup() throws Exception
+    {
+        ServerConnector connector = new ServerConnector(server);
+        server.addConnector(connector);
+
+        FS.ensureEmpty(testDir);
+        FS.ensureEmpty(tmpDir);
+        FS.ensureEmpty(webapps);
+
+        webAppProvider.setMonitoredDirName(webapps.toString());
+        DeploymentManager deploymentManager = new DeploymentManager();
+        deploymentManager.addAppProvider(webAppProvider);
+        server.addBean(deploymentManager);
+
+        HandlerCollection handlerCollection = new HandlerCollection();
+        handlerCollection.addHandler(contexts);
+        handlerCollection.addHandler(new DefaultHandler());
+        deploymentManager.setContexts(contexts);
+        server.setHandler(handlerCollection);
+    }
+
+    @AfterEach
+    public void stop() throws Exception
+    {
+        server.stop();
+    }
+
+    @Test
+    public void testTmpDirectory() throws Exception
+    {
+        Path warPath = MavenTestingUtils.getTestResourcePath("webapps/foo-webapp-1.war");
```

Comment:
The name of the war file and the name of the context xml file must be the same, otherwise the deployer thinks they are 2 different deployables.

Reply:
`warPath` is not pointing to something inside of the monitored directory. So it doesn't really matter what the name is.